core/validatorapi/router.go

```diff
@@ -562,8 +572,19 @@ func proposeBlindedBlock(p eth2client.BlindedBeaconBlockProposalProvider) handle
 
 func submitBlock(p eth2client.BeaconBlockSubmitter) handlerFunc {
 	return func(ctx context.Context, _ map[string]string, _ url.Values, body []byte) (interface{}, error) {
+		capellaBlock := new(capella.SignedBeaconBlock)
```

Comment:
please make sure we have a test for this, ie, when submitting a bellatrix block it gets marshalled as a bellatrix block, etc.

Reply:
I have added a test in `sigagg_test.go`(https://github.com/ObolNetwork/charon/blob/a8c623751c9bb5fa1330b6f90673e350ff71f686/core/sigagg/sigagg_test.go#L262). This tests marshalling & unmarshalling as it calls `block.clone()` which involves both steps.

Reply:
this code isn't tested by that package...

Reply:
we need to test the "submitBlock" function specifically